Assets/Scripts/Extentions/VectorExtensions.cs

```diff
@@ -543,9 +581,11 @@ public static Vector2 ChangeY(this Vector2 v, float y)
         /// <param name="v">This vector2.</param>
         /// <param name="f">The f Vector2.</param>
         /// <returns>A new Vector2 that is a combination.</returns>
-        public static Vector2 CombineVector(this Vector2 v, Vector2 f)
+        public static ref Vector2 CombineVector(ref this Vector2 v, Vector2 f)
```

Comment:
The word combine doesn't always have to mean you're only performing addition. I suggest not overcomplicating the naming and just calling this "Add()"